contracts/governance/StakingRewards/StakingRewards.sol

```diff
@@ -166,8 +207,12 @@ contract StakingRewards is StakingRewardsStorage, Initializable {
 		}
 
 		for (uint256 i = lastWithdrawalInterval; i < duration; i += TWO_WEEKS) {
-			referenceBlock = lastFinalisedBlock.sub(((currentTS.sub(i)).div(32)));
-			if (referenceBlock < deploymentBlock) referenceBlock = deploymentBlock;
+			if (i < upgradeTime) {
+				referenceBlock = lastFinalisedBlock.sub(((currentTS.sub(i)).div(32)));
```

Comment:
`.div(32)` looks very hacky

Reply:
This will hardly be used after this upgrade (i < upgradeTime). We have now linked Staking with StakingRewards and hence can trigger rewards calculation for every staking activity.